<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="page">
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>167</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="2">
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="1">
        <widget class="QWidget" name="widget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="start_pb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>START</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="1">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label">
         <property name="minimumSize">
          <size>
           <width>400</width>
           <height>200</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>400</width>
           <height>200</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="text">
          <string>welcome</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>167</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="1">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_2">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>40</height>
          </size>
         </property>
         <widget class="QPushButton" name="back_p1_pb">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>0</y>
            <width>71</width>
            <height>28</height>
           </rect>
          </property>
          <property name="text">
           <string>BACK</string>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="text">
          <string>choose your data structure</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <property name="minimumSize">
          <size>
           <width>400</width>
           <height>300</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>233333</width>
           <height>233333</height>
          </size>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QPushButton" name="pb1">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>二叉查找树</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="pb2">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>平衡二叉树</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="pb3">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>堆</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="pb4">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>线段树</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pb5">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>主席树</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pb6">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>AC自动机</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="pb7">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="pb8">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="pb9">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="pb10">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QPushButton" name="pb11">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QPushButton" name="pb12">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>80</height>
             </size>
            </property>
            <property name="text">
             <string>NULL</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" colspan="4">
        <widget class="QWidget" name="widget_5" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>40</height>
          </size>
         </property>
         <widget class="QPushButton" name="back_p2_pb">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>0</y>
            <width>71</width>
            <height>28</height>
           </rect>
          </property>
          <property name="text">
           <string>BACK</string>
          </property>
         </widget>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QPlainTextEdit" name="input_text"/>
       </item>
       <item row="2" column="2">
        <widget class="QWidget" name="widget_4" native="true">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="do_pb">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>100</height>
             </size>
            </property>
            <property name="text">
             <string>start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="example_pb">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>100</height>
             </size>
            </property>
            <property name="text">
             <string>use example</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QLabel" name="example_lab">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>120</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="text">
          <string>example</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="4">
        <widget class="QLabel" name="describe_lab">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>500</width>
           <height>200</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="text">
          <string>describe data structure</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
